hathor/debug_resources.py

```diff
@@ -0,0 +1,223 @@
+
+    def run(self, exc_cls: BaseException, msg: str) -> None:
+        deferred = defer.Deferred()
+        deferred.errback(exc_cls(msg))  # type: ignore
+
+
+@register_resource
+class DebugPrintResource(resource.Resource):
+    isLeaf = True
+    openapi = {
+        '/_debug/print': {
+            'x-visibility': 'private',
+            'get': {
+                'operationId': 'debug_print',
+                'summary': 'Used for forcing any print',
+            }
+        }
+    }
+    default_msg = 'debugging python print'
+
+    def render_GET(self, request):
+        bargs = request.args.get(b'msg')
+        if bargs is not None:
+            args = [a.decode() for a in bargs]
+        else:
+            args = [self.default_msg]
+        if b'stderr' in request.args:
+            print(*args, file=sys.stderr)
+        else:
+            print(*args)
+        return b'OK'
+
+
+@register_resource
+class DebugLogResource(resource.Resource):
+    isLeaf = True
+    openapi = {
+        '/_debug/log': {
+            'x-visibility': 'private',
+            'get': {
+                'operationId': 'debug_log',
+                'summary': 'Used for forcing any log message',
+            }
+        }
+    }
+    valid_log_levels = {
+        'critical',
+        'debug',
+        'error',
+        'exception',
+        'fatal',
+        'info',
+        'msg',
+        'warn',
+        'warning',
+    }
+    default_log_level = 'info'
+    default_log_msg = 'debugging logging system'
+
+    def render_GET(self, request):
+        logger_name = get_arg_default(request.args, 'logger', None)
+        if logger_name is not None:
+            log = get_logger(logger_name).new()
+        else:
+            log = logger.new()
+        level = get_arg_default(request.args, 'level', self.default_log_level)
+        assert level in self.valid_log_levels
+        log_func = getattr(log, level)
+        msg = get_arg_default(request.args, 'msg', self.default_log_msg)
+        # TODO: maybe add an `extras` param (probably as a json body via POST) to add arbitray structed attributes
+        log_func(msg)
+        return b'OK'
+
+
+@register_resource
+class DebugMessAroundResource(resource.Resource):
+    isLeaf = True
+    openapi = {
+        '/_debug/mess_around': {
+            'x-visibility': 'private',
+            'get': {
+                'operationId': 'debug_mess_around',
+                'summary': 'Used for forcing an error on the full-node by messing with stuff',
+            }
+        }
+    }
+    default_mess = 'storage'
+
+    def __init__(self, manager: HathorManager):
+        super().__init__()
+        self.manager = manager
+        # map is create here to be able to access the bound methods
+        self.mess_map = {
+            'storage': self.remove_storage,
+        }
+
+    def remove_storage(self):
+        # XXX: this is just used to cause a problem on another part of the fullnode
+        self.manager.tx_storage = None
```

Comment:
Should we an API to revert it?

Reply:
Maybe but I don't think we should expect things to go back to normal after reverting this, so I don't know.

Reply:
You're right. The consequences are unpredictable and the full node must be restarted.